src/lib/http/HttpManager.ts

```diff
@@ -64,49 +66,48 @@ export class HttpClient {
       );
     }
 
-    const res = await axios.post(
+    const response = await axios.post(
       this.tokenURL,
       new URLSearchParams({
         grant_type: 'refresh_token',
         refresh_token: this.config.refreshToken
       }),
       {
-        headers: {
-          authorization: `Basic ${Buffer.from(
-            `${this.config.clientCredentials.clientId}:${this.config.clientCredentials.clientSecret}`
-          ).toString('base64')}`
+        auth: {
+          username: this.config.clientCredentials.clientId,
+          password: this.config.clientCredentials.clientSecret
         },
         validateStatus: () => true
       }
     );
 
-    if (res.status !== 200) {
-      if (res.status === 400) {
-        throw new AuthError(`Refreshing token failed: Bad request`, { data: res.data });
-      }
+    const { status: statusCode } = response;
 
-      if (res.status >= 500 && res.status < 600) {
-        throw new AuthError(`Refreshing token failed: server error (${res.status})`);
-      }
+    if (statusCode === 200) {
+      return response.data.access_token;
+    }
 
-      if (retryAmount < 5) {
-        if (this.config.debug) {
-          console.log(`Refreshing token failed (${res.status}). Retrying... (${retryAmount + 1})`);
-        }
-        await this.refreshToken(retryAmount + 1);
-      } else {
-        throw new AuthError(`Refreshing token failed (${res.status})`);
-      }
+    if (statusCode === 400) {
+      throw new AuthError('Failed to refresh token: bad request', {
+        data: response.data
+      });
     }
 
-    this.config.accessToken = res.data.access_token;
+    if (retryAmount === 5) {
+      if (statusCode >= 500 && statusCode < 600) {
```

Comment:
just reverted back changes made in one of the recent refactoring https://github.com/statsfm/spotify.js/commit/4d6aeb9e668e128e0e6badc658ff424f26507fdb#diff-7a66a64f88500ed3f05b25e049670886f374899bcf5bfcd1a850c19ebaeff100L86-L92 because they cause the token update to fail after the first 5xx response status